e2e/cypress/integration/notifications/unread_on_public_channel_spec.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @notifications
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Notifications', () => {
+    let testUser;
+    let otherUser;
+    let testTeam;
+    let testChannel;
+
+    beforeEach(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+            testTeam = team;
+
+            cy.apiCreateUser().then(({user: user1}) => {
+                otherUser = user1;
+
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id);
+            });
+
+            cy.apiLogin(testUser).then(() => {
+                Cypress._.times(40, (i) => {
+                    // # Create new test channel
+                    cy.apiCreateChannel(testTeam.id, `channel-test${i}`, `Channel${i}`).then((channelRes) => {
+                        if (i === 1) {
+                            testChannel = channelRes.body;
+                        }
+                        cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+                    });
+                });
+
+                // # Most page of messages so the channel can be scrolled up
+                Cypress._.times(40, (i) => {
+                    cy.postMessageAs({sender: otherUser, message: i, channelId: testChannel.id});
```

Comment:
Can you confirm that this is passing on you? I'm getting error here (see screenshot) and may need to rearrange the logic.
<img width="694" alt="Screen Shot 2020-09-02 at 10 29 09 AM" src="https://user-images.githubusercontent.com/5334504/91925324-567fee00-ed07-11ea-84f6-a60372ed5f5b.png">

Reply:
@saturninoabril I have updated the test case, it seems my I pushed out the buggy code.